src/components/network-modification-tree.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { useCallback, useState } from 'react';
+import ReactFlow from 'react-flow-renderer';
+import NetworkModificationNode from './graph/nodes/network-modification-node';
+import ModelNode from './graph/nodes/model-node';
+import CreateNodeMenu from './graph/menus/create-node-menu';
+import NodeEditor from './graph/menus/node-editor';
+import { Box } from '@material-ui/core';
+import { createTreeNode, deleteTreeNode } from '../utils/rest-api';
+
+const nodeTypes = {
+    ROOT: NetworkModificationNode,
+    NETWORK_MODIFICATION: NetworkModificationNode,
+    MODEL: ModelNode,
+};
+
+// snapGrid value set to [15, 15] which is the default value for ReactFlow
+// it has to be explicitly set as prop of the ReactFlow component, even if snapToGrid option is set to false
+// in order to avoid unwanted tree nodes rendering (react-flow bug ?)
+const snapGrid = [15, 15];
+
+const NetworkModificationTree = (props) => {
+    const [selectedNode, setSelectedNode] = useState(null);
+
+    const style = {
+        width: '100%',
+        height: '100%',
+    };
+
+    const onSelectionChange = useCallback((selectedElements) => {
+        if (selectedElements?.length > 0) {
+            setSelectedNode(selectedElements[0]);
+        }
+    }, []);
+
+    const onPaneClick = useCallback(() => {
+        setSelectedNode(undefined);
+    }, []);
+
+    const handleCreateNode = useCallback((element, type) => {
+        createTreeNode(element.id, { name: 'New node', type: type }).then(
+            (response) => {
+                if (response.status !== 200) {
+                    console.log('Error creating node');
+                }
+            }
+        );
+    }, []);
+
+    const handleRemoveNode = useCallback((element) => {
+        deleteTreeNode(element.id).then((response) => {
+            if (response.status !== 200) {
```

Comment:
snackbar

Reply:
Error message is now displayed in snackbar